### PR TITLE
Bootstrap web3-gateway

### PR DESCRIPTION
**Pull request by edisonz0718**
issue #117 

Tested in staging:
Screen Shot 2023-05-22 at 9 35 33 PM

### CODE REVIEW

File: dockerfile/builder/Dockerfile [hunk 1]
```diff
@@ -7,13 +7,27 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
...
+    unzip -o $PROTOC_ZIP -d /usr/local bin/protoc && \
+    unzip -o $PROTOC_ZIP -d /usr/local 'include/*' && \
+    rm -f $PROTOC_ZIP
 
 # Golang related.
 # https://maelvls.dev/go111module-everywhere/
-ENV GO111MODULE=on
\ No newline at end of file
+ENV GO111MODULE=on
+RUN go install github.com/golang/protobuf/protoc-gen-go@latest && \
```
Comment (LeoHChen) [inline dockerfile/builder/Dockerfile:30]: better use specific version for protoc, latest may not be backward compatible in some cases, and you are using protoc 3.7.1

File: dockerfile/builder/Dockerfile [hunk 2]
```diff
@@ -7,13 +7,27 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
...
+    go install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger@latest
```
Comment (LeoHChen) [inline dockerfile/builder/Dockerfile:31]: same thing for swagger, better use a specific version

File: dockerfile/builder/Dockerfile [hunk 3]
```diff
@@ -7,13 +7,27 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
...
+
+RUN npm install -g typescript@latest
```
Comment (LeoHChen) [inline dockerfile/builder/Dockerfile:33]: please use a specific typescript version

File: web3-gateway/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Web3 gateway is the gatway layer for web3 services
```
Comment (LeoHChen) [inline web3-gateway/README.md:1]: type "gatway"

**Review by LeoHChen — APPROVED** [review]
LGTM. Please add the design doc of web3gateway to the readme doc in the repo.

**Comment by edisonz0718** [comment]
> LGTM. Please add the design doc of web3gateway to the readme doc in the repo.

Will add in a separate PR. #119

**Pull request merged**